demisto_sdk/commands/update_xsoar_config_file/tests/update_xsoar_config_file_test.py

```diff
@@ -70,6 +70,40 @@ def test_add_all_marketplace_packs(self, mocker, add_all_marketplace_packs, expe
                 config_file_info = {}
             assert config_file_info == expected_outputs
 
+    def test_add_all_marketplace_packs_on_existing_list(self, mocker):
+        """
+        Given:
+            - add_all_marketplace_packs arg as True
+        When:
+            - run the update_xsoar_config_file command
+        Then:
+            - validate the xsoar_config file exist in the destination output
+            - validate the xsoar_config file output is as expected
+        """
+
+        mocker.patch.object(XSOARConfigFileUpdater, 'get_installed_packs', return_value=[
+            {"id": "test1", "version": "1.0.0"}])
+        with temp_dir() as tmp_output_dir:
+
+            with open(f'{tmp_output_dir}/xsoar_config.json', 'w') as config_file:
+                json.dump({'marketplace_packs': [{'id': 'test2', 'version': '2.0.0'}]}, config_file)
+
+            click.Context(command=xsoar_config_file_update).invoke(
+                xsoar_config_file_update,
+                file_path=tmp_output_dir / 'xsoar_config.json',
+                add_all_marketplace_packs=True
+            )
+
+            assert Path(f'{tmp_output_dir}/xsoar_config.json').exists()
+
+            try:
+                with open(f'{tmp_output_dir}/xsoar_config.json', 'r') as config_file:
+                    config_file_info = json.load(config_file)
+            except IsADirectoryError:
+                config_file_info = {}
+            assert config_file_info == {'marketplace_packs': [{'id': 'test2', 'version': '2.0.0'},
+                                                              {'id': 'test1', 'version': '1.0.0'}]}
+
```

Comment:
it may be easier to see the diff [here](https://github.com/demisto/demisto-sdk/pull/1999/files) (branch to be merged into this one)

```suggestion
    def test_add_all_marketplace_packs(self, mocker,
                                       add_all_marketplace_packs: bool,
                                       expected_path: str,
                                       expected_outputs: dict):
        """
        Given:
            - add_all_marketplace_packs arg as True or False
        When:
            - run the update_xsoar_config_file command
        Then:
            - validate the xsoar_config file exist in the destination output
            - validate the xsoar_config file output is as expected
        """

        mocker.patch.object(XSOARConfigFileUpdater, 'get_installed_packs', return_value=[
            {"id": "test1", "version": "1.0.0"}])
        with temp_dir() as tmp_output_dir:
            temp_file = tmp_output_dir / 'xsoar_config.json'
            click.Context(command=xsoar_config_file_update).invoke(
                xsoar_config_file_update,
                file_path=temp_file,
                add_all_marketplace_packs=add_all_marketplace_packs
            )
            assert temp_file.exists() == add_all_marketplace_packs

            expected_path_object = Path(tmp_output_dir) / expected_path
            if expected_path_object.is_file():
                config_file_info = json.loads(expected_path_object.read_text())
            elif expected_path_object.is_dir():
                config_file_info = {}
            else:
                raise ValueError(f"Cannot read {str(expected_path_object)}")
            assert config_file_info == expected_outputs

```

Reply:
done